pkgs/by-name/si/siyuan/package.nix

```
@@ -20,6 +22,15 @@ let
   };
 
   platformId = platformIds.${stdenv.system} or (throw "Unsupported platform: ${stdenv.system}");
+
+  desktopEntry = makeDesktopItem {
+    name = "siyuan";
+    desktopName = "Siyuan";
```

```suggestion
    desktopName = "SiYuan";
```

Not very important, but upstream also capitalizes the letter Y.

Here's upstream's .desktop file:
```
[Desktop Entry]
Name=SiYuan
Exec=/opt/SiYuan/siyuan %U
Terminal=false
Type=Application
Icon=siyuan
StartupWMClass=SiYuan
Comment=Refactor your thinking
Categories=Utility;
```

If you want to, you can change the category / comment too. Those being different is not a huge deal, so it's fine to leave it as is IMO.